in_toto/runlib.py

```diff
@@ -315,20 +318,26 @@ def create_link_metadata(link_name, materials_dict=None, products_dict=None,
   if not byproducts:
     byproducts = {}
 
+  if record_working_directory:
+    working_directory = os.getcwd()
+  else:
+    working_directory = ""
+
   link_dict = {
     "name" : link_name,
     "materials" : materials_dict,
     "products" : products_dict,
     "command" : link_cmd_args,
     "byproducts" : byproducts,
-    "return_value" : return_value
+    "return_value" : return_value,
+    "working_directory" : working_directory
```

Comment:
If you mean to conditionally assign the attribute in `Link`, that would not change anything, each object will sill always have that attribute, and in consequence the (empty) field in the JSON.

You could ignore the field when converting the class to JSON. But (if you want to do it properly) you'd have to write something like [`attr`'s as_dict](https://github.com/in-toto/in-toto/blob/develop/in_toto/models/common.py#L40-L41) function which currently does this job. 

Which on the other hand wouldn't be too bad, because [it's not the huge](https://github.com/python-attrs/attrs/blob/1b419e304b01bbbcb1d0d7b0515efdbc7de73588/src/attr/_funcs.py#L10-L67) and above all that function is pretty much the only thing that currently keeps us from removing that dependency.